aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Orchestrator.java

```diff
@@ -275,12 +289,7 @@ private void startObservingStorageChanges() throws DataStoreException {
         }
     }
 
-    /**
-     * Stop observing the local storage. Do not enqueue changes to the outbox.
-     */
-    private void stopObservingStorageChanges() {
-        LOG.info("Stopping observation of local storage changes.");
-        storageObserver.stopObservingStorageChanges();
```

Comment:
Great question! After taking another look I noticed that this part of the changeset is not needed, because this code path is only executed when `DataStore.stop()` is called and and that point it is ok to stop listening for local changes.

I'll be reverting this part of the PR